### PR TITLE
Android Runtime Banner - Chrome Intent support

### PR DESCRIPTION
**Pull request by fdoxyz**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

Runtime Banner Android app support 

## QA Instructions, Screenshots, Recordings

- Open the app using iOS -> browse site and click on banner -> Should deep link into mobile app
- Open the app using Android -> browse site and click on banner -> Should deep link into mobile app

### UI accessibility concerns?

None - UI hasn't changed

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: The Android mobile app will be communicated accordingly in sync with the Marketing team

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by citizen428** [comment]
Ho @fdoxyz and @rt4914, what's the status of this PR? Will we eventually merge it?

**Comment by rt4914** [comment]
@citizen428 This PR can only be merged once the app is open-beta or production. Mostly the app will be in open-beta by end of this week.

**Reason**: Once this PR gets merged everyone who opens any forem in their android-mobile-browser will see a runtime banner at the bottom of screen which when clicked will redirect to playstore to install the app or directly to the app if its already install. Currently the playstore link is not public and therefore we cannot merge this PR.

**Comment by rt4914** [comment]
@fdoxyz This is ready to merge now. The app is in open-beta now.

**Comment by fdoxyz** [comment]
@rt4914 I was having trouble making this one work locally, now that's sync'ed with the `main`. Maybe we can give this one a try on a Canary instance to double check it works before rolling it out.

**Comment by andygeorge** [comment]
`quay.io/forem/forem:pr-15181` is now deployed on https://canary1.forem.wtf

**Comment by andygeorge** [comment]
`quay.io/forem/forem:pr-15181` is now deployed on Canary1

File: app/javascript/runtimeBanner/RuntimeBanner.jsx
Comment (nickytonline) [inline app/javascript/runtimeBanner/RuntimeBanner.jsx:12]: Refactor (suggestion): These can be environment variables that webpack can inject if it makes sense. If these never/rarely change, keeping them as constants is fine.

**Review by nickytonline — COMMENTED** [review]
I don't have an Android app to test this, but the code looks good. I made one suggestion.

**Comment by rt4914** [comment]
@fdoxyz Just install the latest version of the app `1.0.2` v8. It should work now.

File: app/javascript/runtimeBanner/RuntimeBanner.jsx
Comment (fdoxyz) [reply app/javascript/runtimeBanner/RuntimeBanner.jsx:12]: Thanks for the suggestion @nickytonline! Yup these AppStore/Play Store links aren't supposed to ever change so we should be safe with this approach then 👍

**Review by rt4914 — APPROVED** [review]
I have checked and this PR is working correctly. Thanks @fdoxyz

**Comment by fdoxyz** [comment]
Actually just took down the `merge by any core` label. I'm thinking it will be best if we make sure someone from the team has an Android device and is able to test and ensure the deployment goes smoothly. Given that it has a big UX impact we should revert if needed. I'm confident this won't be the case, but it doesn't hurt to be extra cautious  😃

**Review by Zhao-Andy — APPROVED** [review]
QA'ed and worked for me on Android Chrome!

**Comment by fdoxyz** [comment]
After trying out different ideas I've disabled the banner on Firefox for Android because of the bug we were noticing.

I created an issue in the [Firefox browser repo here](https://github.com/mozilla-mobile/fenix/issues/23397) to ensure this isn't a problem with out Intent, in case it actually is a bug on the browser handling the Intent link.

This PR is ready to go as soon as we double check we're not displaying the Banner on Firefox.

**Pull request merged**
